028-service-binding.md

```diff
@@ -0,0 +1,276 @@
+```
+
+Naming of secret keys:
+
+This proposal previously considered choosing secret keys that could be reused by other (non-Kafka) operators that use mutualTLS or SASL SCRAM authentication. However, given 
+that Kafka applications generally have a well known list of configuration options they provide, the proposal instead uses those well-known keys. This will make it clearer 
+to application developers what the different keys should be used for.
+
+### Implementation
+
+This proposal will introduce a new `KafkaAccess` operator. The operator will be separate from the existing `strimzi-kafka-operator` and live in its own GitHub repo in the `strimzi` organization.
+
+The actions taken during the reconcile loop of the new operator will be:
+1. The operator can be deployed at cluster-scope where it will watch resources across all namespaces, or in a single namespace, where it will only watch resources in that namespace.
+2. On start-up the operator will watch all resources of Kind `KafkaAccess`.
+3. For any `KafkaAccess` resource present it will add a status of NotReady to that CR.
+4. For each `KafkaAccess` resource it will perform a GET request on the Kafka cluster that is 
+referenced in the `KafkaAccess` and pick out the configured listeners.
+   1. If this or any subsequent GET requests fail due to a 401 (unauthorised) error, the status reason will be updated to reflect this.
+5. The operator will determine which listener to choose:
+   1. If there are no listeners in the `Kafka` CR the reconcile loop will fail.
+   2. If the `KafkaAccess` specifies a listener, the operator will choose this listener.
+   3. If there are no listeners specified in the `KafkaAccess`, but only one listener in the `Kafka` CR, the operator will choose that listener.
+   4. If there are no listeners specified in the `KafkaAccess`, but multiple listeners listed in the `Kafka` CR, the operator 
+   will filter the list by comparing the `tls` and `authentication` properties in the `Kafka` and `KafkaUser` CRs.
+   5. If there are still multiple listeners after filtering for `tls` and `authentication` the operator will choose the one that is of type `internal`.
+   6. If there are still multiple listeners after filtering for `internal` the operator will sort the listeners alphabetically by name and choose the first one.
+   7. If the operator fails to find a suitable listener after filtering the reconcile loop will fail.
+8. The operator will then create a secret and add the bootstrapServers for the chosen listener to that secret.
+9. If the chosen listener uses TLS, the operator will then perform a GET request on the `cluster-ca-cert` secret and put the certificate and truststore files in the new secret.
+10. If the `KafkaAccess` references a `KafkaUser`, the operator will perform a GET request on that user to find the name of the secret containing the credentials and the username 
```

Comment:
You need to have the user already before the step 5.4 -> so this needs to be done earlier.

Reply:
Good point, I have it earlier in my POC